lib/macho/macho_file.rb

```diff
@@ -363,13 +363,16 @@ def rpaths
     # @param new_path [String] the new runtime path
     # @return [void]
     # @raise [MachO::RpathUnknownError] if no such old runtime path exists
+    # @raise [MachO::RpathExistsError] if the new runtime path already exists
     def change_rpath(old_path, new_path)
       old_lc = command(:LC_RPATH).find { |r| r.path.to_s == old_path }
       raise RpathUnknownError.new(old_path) if old_lc.nil?
+      raise RpathExistsError.new(new_path) if rpaths.include?(new_path)
 
       new_lc = LoadCommand.create(:LC_RPATH, new_path)
 
-      replace_command(old_lc, new_lc)
+      delete_rpath(old_path)
+      insert_command(old_lc.view.offset, new_lc)
```

Comment:
Yes, either that or we also start making this assumption in `delete_rpath`. The latter option seems also reasonable to me, but for me to accept it I'd like to see that documented (make the assumed sort order official) and enforced in tests (make sure the order stays the same if we end up refactoring some bits). There's of course an advantage in avoiding the sort if we know the array is always pre-sorted.

Pick your favorite option. 😸 Independently of the option (both are valid), I tend to prefer a defensive coding style for operations like this one. Otherwise we'll easily get corrupted Mach-O binaries if something unexpectedly or accidentally changes in the code base.